[#168] feat: Remove placeholder data

IanSeng opened the pull request:
# Summary 
This pr is to remove placeholder values for the author's name, word count, and copyright as well as enhancing the input field by adding the feature dynamically update the subtext when user input. 
By leveraging the feature build in [#147](https://github.com/eddieantonio/predictive-text-studio/pull/169) it allows this pr to further simplify the logic in `src/app/components/BCP47Tag.svelte` by relying a more reusable component where child component is now able to communicate with parents components. 

**Note**: this pr has dependency on [#147](https://github.com/eddieantonio/predictive-text-studio/pull/169) because I needed the feature built in #147 to fix `BCP47Tag.svelte component`. The pr review order should ideally in the sequence of [#147](https://github.com/eddieantonio/predictive-text-studio/pull/169)> [This PR](https://github.com/eddieantonio/predictive-text-studio/pull/177) > [#165](https://github.com/eddieantonio/predictive-text-studio/pull/178). Feel free to ping me if there is any questions ☺️

Feel free to ping me if there is any questions ☺️

## Sub-tasks 
- [x] Remove placeholder data
- [x] Fix BCP47Tag.svelte component
- [x] Add dynamic value binding to input 
- [x] Remove used code
- [x] Update existing e2e test
 
## Issue
Resolves #168

IanSeng committed:
fix on input

IanSeng committed:
fix bcp47tag component

IanSeng committed:
remove uncessary metadata

IanSeng committed:
code clean up

IanSeng committed:
code restructure

IanSeng committed:
fix bcp47tag logic

IanSeng committed:
update e2e to use data

xqiu1 reviewed (approved):
LGTM

eddieantonio commented on src/app/components/InputField.svelte:
```diff
@@ -79,12 +87,13 @@
+    on:input={dispatchInputValueOnInput} />
   {#if error !== ''}
-    <p class:error>{error}</p>
+    <p class:error data-cy="input-error-message">{error}</p>
```
This makes the tests cases a lot more robust with respect to changes! 👍🏼

eddieantonio commented on src/app/components/BCP47Tag.svelte:
```diff
@@ -3,22 +3,27 @@
-  $: error = validTag ? "" : "Invalid BCP 47 Tag";
-  $: if (validTag) {
-    worker.setProjectData({ languages: [{ name: "", id: tag }] });
+  function onInputValue(event: CustomEvent) {
```
Could be called something like... this?

```suggestion
  function validateAndSetLanguage(event: CustomEvent) {
```

eddieantonio commented on src/app/components/InputField.svelte:
```diff
@@ -9,19 +9,27 @@
+  function dispatchInputValueOnInput(event: Event) {
+    const target = event.target as HTMLTextAreaElement;
+    const cleanId = target.id.split("-").pop();
+    dispatch("messageOnChange", {
```
I wonder if there's a better name for this event. The camel case is weird in the message name, since I don't think many DOMEvent names have camel case at all.
https://developer.mozilla.org/en-US/docs/Web/Events

Distressingly, the most common event name convention is `abunchofsmooshedtogetherwords` 😩 

Maybe...

```suggestion
    dispatch("keytyped", {
```

eddieantonio reviewed (approved):
LGTM, but I have a few vague naming suggestions!

IanSeng commented:
> LGTM, but I have a few vague naming suggestions!

Thanks @eddieantonio , I will spend some time right after my finals to address all of the feedbacks. Sorry for the delay 🤓

IanSeng committed:
address feedback

IanSeng commented:
Addressed all comments accordingly. 
![651d5042e49e2eea63ea94c09e119716_w200](https://user-images.githubusercontent.com/35438972/102824517-3ea5a600-439a-11eb-9cbc-056edfd88733.gif)
